### PR TITLE
[14.0] [ADD] account_invoice_restrict_linked_so: New module to restrict invoice lines created from SO.

### PR DESCRIPTION
**Pull request by Chandresh-OSI**

### CODE REVIEW

File: account_invoice_restrict_linked_so/views/account_move.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            <xpath expr="//field[@name='invoice_line_ids']/tree" position="inside">
+                <field name="from_so_line" invisible="1" />
+            </xpath>
+            <xpath
+                expr="//field[@name='invoice_line_ids']/tree/field[@name='product_id']"
+                position="attributes"
+            >
+                <attribute
+                    name="attrs"
+                >{'readonly':[('from_so_line','!=',False)]}</attribute>
```
Comment (dreispt) [inline account_invoice_restrict_linked_so/views/account_move.xml]: The `from_so_line` field bring little benefit.
You can just check if `sale_line_ids` is set or empty.

```suggestion
                >{'readonly':[('sale_line_ids','!=',False)]}</attribute>
```

**Review by dreispt — COMMENTED** [review]
Code review looks good. Will do some tests.

**Review by dreispt — APPROVED** [review]
Tested, works as expected.

**Review by patrickrwilson — APPROVED** [review]
LGTM

File: account_invoice_restrict_linked_so/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    "name": "Restrict Invoice created from SO",
+    "summary": "Restricts editing the Product, Quantity and Unit Price "
+    "columns for invoice lines that originated in Sales Orders.",
+    "version": "14.0.1.0.1",
+    "category": "Accounting",
+    "author": "Open Source Integrators, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/account-invoicing",
+    "license": "AGPL-3",
+    "depends": ["sale_management"],
```
Comment (bodedra) [inline account_invoice_restrict_linked_so/__manifest__.py:13]:
```suggestion
    "depends": ["account", "sale_management"],
```

**Review by bodedra — APPROVED** [review]
Functional reviewed. :+1: 

Tried with normal sale invoice and down payment. Works great!

Added suggestion to add `account` module in a dependency.

File: account_invoice_restrict_linked_so/__manifest__.py (hunk 2 above)
Comment (Chandresh-OSI) [reply account_invoice_restrict_linked_so/__manifest__.py:13]: No need for that. `sale_management` is depends on `sale`.
`payment` module is in depends of `sale` module.
`account` module is in depends of `payment` module.

**Pull request merged**